services/cd-service/pkg/repository/transformer.go

```diff
@@ -374,9 +373,79 @@ func (c *CreateApplicationVersion) Transform(ctx context.Context, state *State)
 			result = result + deployResult + "\n"
 		}
 	}
+	err = writeCommitData(ctx, c.SourceCommitId, c.Application, c.EventId, allEnvsOfThisApp, fs)
+	if err != nil {
+		return "", nil, GetCreateReleaseGeneralFailure(err)
+	}
+
 	return fmt.Sprintf("created version %d of %q\n%s", version, c.Application, result), changes, nil
 }
 
+func writeCommitData(ctx context.Context, sourceCommitId string, app string, eventId string, environments []string, fs billy.Filesystem) error {
+	if !valid.SHA1CommitID(sourceCommitId) {
+		return nil
+	}
+	commitDir := commitApplicationDirectory(fs, sourceCommitId, app)
+	if err := fs.MkdirAll(commitDir, 0777); err != nil {
+		return err
+	}
+	if err := util.WriteFile(fs, fs.Join(commitDir, ".dir"), make([]byte, 0), 0666); err != nil {
+		return err
+	}
+	err := writeEvent(ctx, eventId, sourceCommitId, fs, environments)
+	if err != nil {
+		return fmt.Errorf("error while writing event: %v", err)
+	}
+
+	return nil
+}
+
+func writeEvent(ctx context.Context, eventId string, sourceCommitId string, filesystem billy.Filesystem, envs []string) error {
+	if eventId == "" {
+		fullUuid, err := uuid.NewUUID()
+		if err != nil {
+			return fmt.Errorf("could not create uuid: %v", err)
+		}
+		// format of this "uuid" is the same as the lock IDs, see "randomLockId" in store.tsx
+		eventId = "cd-v1-" + string(fullUuid.String()[:7])
+	}
+	eventDir := commitEventDir(filesystem, sourceCommitId, eventId)
+	_, err := filesystem.Stat(eventDir)
+	if err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			// anything except "not exists" is an error:
+			return fmt.Errorf("event directory %s already existed: %v", eventDir, err)
+		}
+	}
+	if err := filesystem.MkdirAll(eventDir, 0777); err != nil {
+		return fmt.Errorf("could not create directory %s: %v", eventDir, err)
+	}
+	for i := range envs {
```

Comment:
Do we actually need to store a list of all environments here again? Wouldn't it be much more interesting to store the release version?

Reply:
I would like to make the rpc call return all required information to render if possible.
In this case, this list of envs correlates with "which envs have manifests", which I think is a relevant question.
We could add the release version as well, but I don't require it for this story.

Reply:
It's easier for the client like this. We have one call that returns all required data.
We can later add the release version, once we need it.